enhancements/hide-container-mountpoints.md

```diff
@@ -0,0 +1,333 @@
+systemd `oneshot` with no actual process that needs keeping alive, the risk of
+this terminating unexpectedly is very low.
+
+Hiding the Kubernetes mounts from systemd may confuse administrators and
+support personnel who are used to seeing them.
+
+## Design Details
+
+### Test Plan
+
+- With the feature enabled:
+  - Ensure that running 'mount' in default mount namespace does not show any of
+    the Kubernetes-specific mountpoints.
+  - Ensure that entering the mount namespace and running 'mount' shows all the
+    Kubernetes-specific mountpoints.
+  - All existing e2e tests at a similar rate.
+- With the feature disabled:
+  - Ensure that running 'mount' in default mount namespace shows all of the
+    Kubernetes-specific mountpoints.
+  - All existing e2e tests at a similar rate.
+
+### Graduation Criteria
+
+The main graduation consideration for this feature is when it is enabled by
+default.
+
+This feature is already in Dev Preview, with the current MachineConfig-based
+proof-of-concept solution part of the telco-specific DU profile installed by
+ZTP, and also available
+[here](https://github.com/openshift-kni/cnf-features-deploy/tree/master/feature-configs/deploy/container-mount-namespace)
+
+#### Dev Preview -> Tech Preview
+
+- Reimplement according to this proposal, but the feature is disabled by
+  default
+- Add a CI lane that runs all current e2e tests with this feature enabled
+- Update the ZTP DU profile to use the new mechanism instead of the current
+  MachineConfig-based proof-of-concept
+
+#### Tech Preview -> GA
+
+- Enable the feature by default
+- Remove the CI lane that enables the feature, as it is enabled by default
+- User-facing documentation to detail the mechanism to inspect container mounts
+  or disable the feature if needed
+
+#### Removing a deprecated feature
+
+Not applicable.
+
+### Upgrade / Downgrade Strategy
+
+Not applicable.
+
+### Version Skew Strategy
+
+Not applicable. The mount namespace is fully-contained and isolated within each
+node of a cluster. There is no impact of having the feature enabled on some
+nodes and disabled on others.
+
+### Operational Aspects of API Extensions
+
+Not applicable; no API extensions; but there are operational impacts of this
+change, detailed in the 'Risks and Mitigations' section above.
+
+#### Failure Modes
+
+- If either the Kubelet or CRI-O services end up in different namespaces from
+  one another, containers started by CRI-O will not see mounts made by Kubelet,
+  such as secrets or configmaps.
+
+- If the namespace is not configured correctly to allow mounts from the OS to
+  be shared into Kubelet or CRI-O, system mountpoints will not be visible to
+  either Kubelet or CRI-O or runing containers.
+
+#### Support Procedures
+
+When this feature is enabled, a shell on a node and the 'oc debug' container
+will not have visibility of the Kubernetes mountpoints.
+
+- To start a shell within the container mount namespace, execute the
```

Comment:
I think updates to the debug pod MOTD or a wrapper script around the pod's chroot could be helpful